WordPress/src/test/java/org/wordpress/android/viewmodel/pages/PageListViewModelTest.kt

```diff
@@ -186,6 +203,48 @@ class PageListViewModelTest : BaseUnitTest() {
         assertThat((result[0].first[1] as PublishedPage).title).isEqualTo(secondPage.title)
     }
 
+    @Test
+    fun `checks if PageItems gets the correct progress state`() {
```

Comment:
We might want to consider breaking up this test so each test method has only a single reason to fail. We are testing multiple unrelated things within one test method. Wdyt?

Reply:
Okay. Thanks for this. I thought just testing the behavior of the progress state was a single thing. I will revisit 😄

Reply:
Oh, the test is checking the state of two different pages and behavior, right? I am going to refactor and put it into two separate tests. Thanks.

Reply:
Resolved in 175a7c3

Reply:
Awesome, it's much better now. Thank you!

We might want to consider two more changes
1. Moving the most complicated test after the two simpler tests.
2. Mocking things we don't need for the test and creating local fields only from variables we actually need to use in the test

Something like this

```
 @Test
    fun `showOverlay is correctly propagated from PageItemUploadProgressHelper`() {
        // Arrange
        val expectedShowOverlay = true
        val pages = MutableLiveData<List<PageModel>>()

        whenever(progressHelper.getProgressStateForPage(any())).thenReturn(Pair(mock(), expectedShowOverlay))
        whenever(pagesViewModel.pages).thenReturn(pages)

        viewModel.start(PUBLISHED, pagesViewModel)
        val result = mutableListOf<Pair<List<PageItem>, Boolean>>()
        viewModel.pages.observeForever { result.add(it) }
        // Act
        pages.value = listOf(buildPageModel(0))
        // Assert
        assertThat((result[0].first[0] as Page).showOverlay).isEqualTo(expectedShowOverlay)
    }
```

Wdyt?

Reply:
Nice! This improvement is like magic 😉

Reply:
Resolved in aecfb9f